rmw_zenoh_cpp/src/detail/graph_cache.hpp

```diff
@@ -137,20 +122,14 @@ class GraphCache final
   namespace_2:
     node_n:
   */
+
+  using NodeMap = std::unordered_map<std::string, GraphNodePtr>;
+  using NamespaceMap = std::unordered_map<std::string, NodeMap>;
   // Map namespace to a map of <node_name, GraphNodePtr>.
-  std::unordered_map<std::string, std::unordered_map<std::string, GraphNodePtr>> graph_ = {};
-
-  // Optimize topic lookups mapping "topic_name?topic_type" keys to their pub/sub counts.
-  struct TopicStats
-  {
-    std::size_t pub_count_;
-    std::size_t sub_count_;
-
-    // Constructor which initialized counters to 0.
-    TopicStats(std::size_t pub_count, std::size_t sub_count);
-  };
-  using TopicStatsPtr = std::unique_ptr<TopicStats>;
-  std::unordered_map<std::string, TopicStatsPtr> graph_topics_ = {};
+  NamespaceMap graph_ = {};
+
+  // Optimize topic lookups across the graph.
+  GraphNode::TopicMap graph_topics_ = {};
```

Comment:
With the cache, lookups to check if a certain topic exists in the graph have `O(1)` complexity on average. Else if we have to iterate through `graph_cache_`, it will be `O(n*n'*n'')` where we have to iterate through `n` namepaces, `n'` nodes within each namespace and potentially `n''` topics per node.